Save variant metadata after merging variants

arahuja opened the pull request.

tavinathanson commented on cohorts/load.py:
```diff
@@ -489,38 +488,53 @@ def load_variants(self, patients=None, variant_type="snv", merge_type="union"):
             failed_io = True
 
-        if len(combined_variants) == 0 or failed_io:
+        if len(variant_collections) == 0 or failed_io:
```
This will be undefined if there was an exception, no?

arahuja replied:
Ah yea, I see that, will fix, can sub in `vcf_paths`

arahuja replied:
I switched the order of checks here

tavinathanson commented on cohorts/load.py:
```diff
@@ -489,38 +488,53 @@ def load_variants(self, patients=None, variant_type="snv", merge_type="union"):
+        if len(variant_collections) == 1:
             # There is nothing to merge
-            merged_variants =  VariantCollection(combined_variants[0])
+            merged_variants = VariantCollection(variant_collections[0][1].elements)
```
Not obvious to me what the `[0][1]` is? Take the first `VariantCollection`, and then what?

tavinathanson replied:
Oh, I see, `variant_collections` is a list of tuples. Not yet clear why the tuples are necessary?

arahuja replied:
I wanted to hang to the vcf path's as well to know which file the metadata came from

arahuja replied:
@tavinathanson I updated this to actually keep the metadata in the single vcf case as well

tavinathanson commented on cohorts/load.py:
```diff
@@ -489,38 +488,54 @@ def load_variants(self, patients=None, variant_type="snv", merge_type="union"):
 
         return merged_variants
 
+    def _merge_variant_collections(self, variant_collections, merge_type):
```
Confusing that `variant_collections` is a `dict` here and list of tuples above? Maybe rename to `vcf_to_variant_collections` here? Or make it a `dict` throughout?

arahuja replied:
I renamed this `vcf_to_variant_collections` and rebased

tavinathanson commented:
LGTM % one more comment; though seems like `varcode` is a better place for logic to combine `VariantCollection`s? If you agree, maybe file an issue there that references this?

arahuja commented:
I updated the variable name and rebased for the merge conflicts

> though seems like varcode is a better place for logic to combine VariantCollection

Maybe? I'm not sure it is, but I'm indifferent on it

tavinathanson commented:
👍
